python/cudf/cudf/_lib/copying.pyx

```diff
@@ -265,3 +270,52 @@ def copy_range(out_col, in_col, int out_begin, int out_end,
     free_column(c_out_col)
 
     return out_col
+
+
+def scatter_to_frames(source, maps):
+    """
+    Scatters rows to 'n' dataframes according to maps
+
+    Parameters
+    ----------
+    source : Column or list of Columns
+    maps : non-null column with values ranging from 0 to n-1 for each row
+
+    Returns
+    -------
+    list of scattered dataframes
+    """
+    from cudf.core.column import column
+
+    in_cols = source
+    for i, in_col in enumerate(in_cols):
+        in_cols[i] = column.as_column(in_cols[i])
+
+    if in_cols[0].dtype == np.dtype("object"):
+        in_size = in_cols[0].data.size()
+    else:
+        in_size = in_cols[0].data.size
+
+    maps = column.as_column(maps).astype("int32")
+
+    col_count=len(in_cols)
+    gather_count = len(maps)
+    assert(gather_count == in_size)
+
+    cdef gdf_column** c_in_cols = cols_view_from_cols(in_cols)
+    cdef cudf_table* c_in_table = new cudf_table(c_in_cols, col_count)
+    cdef gdf_column* c_maps = column_view_from_column(maps)
+    cdef vector[cudf_table] c_out_tables
+
+    if gather_count != 0:
```

Comment:
Should this be pushed down into the C++?

Reply:
This is already taken care in C++.  removed.